lib/opFns.js

```diff
@@ -157,6 +157,45 @@ module.exports = {
 
     return new BN(word.shrn((31 - pos.toNumber()) * 8).andln(0xff))
   },
+  SHL: function (a, b, runState) {
+    if (!runState._common.gteHardfork('constantinople')) {
+      trap(ERROR.INVALID_OPCODE)
+    }
+    if (a.gten(256)) {
+      return new BN(0)
+    }
+    return b.shln(a.toNumber()).iand(utils.MAX_INTEGER)
```

Comment:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-145.md#0x1b-shl-shift-left

I am not super-fluent on bit operations, but I think I have found all conditions from the EIP in the implementation.

Reply:
(always a bit shaky on this signed/unsigned stuff)